### PR TITLE
Run dbt on WebAssembly using Pyodide

### PR DESCRIPTION
**Pull request by arieldbt**
### Description

Changes needed to run dbt in the browser (using WebAssembly via Pyodide). For more context on the rationale see the [discussion](https://github.com/dbt-labs/docs.getdbt.com/issues/1970) in github and our community slack. Here is a demo of the jaffle shop example project using sqlite and dbt in the browser:

https://s3.amazonaws.com/com.getdbt.cloud.wasm/index.html

To enable running dbt on WebAssembly I needed to refactor two parts of the codebase into clients since they behave differently in a browser compared to a normal operating system:

1. HTTP requests
2. Parallel processing

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by jtcohen6** [comment]
@cla-bot check

**Comment by ChenyuLInx** [comment]
@arieldbt  looks great! I probably will have more time to play with it in October!

File: core/setup.py [hunk 1]
```diff
@@ -29,6 +29,34 @@
...
+    "idna>=2.5,<4",
+    "cffi>=1.9,<2.0.0",
+    "pyyaml>=6.0",
+]
+
+if "DBT_WASM_BUILD" in os.environ and int(os.environ["DBT_WASM_BUILD"]) == 1:
+    # binary dependency not supported in pyodide
+    pass
+else:
+    _install_requires.insert(14, "dbt-extractor~=0.4.1")
```
Comment (jtcohen6) [inline core/setup.py:59]: Is there any cleaner way for us to manage this? Would this add another point of complexity that would prevent us from moving to `pyproject.toml` in the future? (https://github.com/dbt-labs/dbt-core/issues/4674)

**Comment by gshank** [comment]
Ariel has moved on, so unless he wants to contribute to this as an external contributor...

I've run pre-commit on it, and there are a number of complaints from mypy. I didn't look closely at all of them, but I think the code in flags.py for Pyodide should probably go somewhere else at least.

**Comment by ajmarcus** [comment]
👋  - been a bit busy but happy to see if I can help with the pre-commit / code-quality checks.

Probably should have had the pre-commit hooks already set up on my machine. My bad.

**Comment by jtcohen6** [comment]
Closing in favor of https://github.com/dbt-labs/dbt-core/pull/6146

**Pull request closed without merging**